src/core/Core.js

```diff
@@ -1084,8 +1084,14 @@ class Uppy {
           waitingFileIDs.push(file.id)
         }
       })
-
-      const uploadID = this._createUpload(waitingFileIDs)
+      const { currentUploads } = this.getState()
+
+      // get a list of files that are currently assigned to uploads
+      const currentlyUploadingFiles = Object.keys(currentUploads).reduce((prev, curr) => prev.concat(currentUploads[curr].fileIDs), [])
+      // files the list of files that are waiting to be uploaded against ones that are assigned to uploads. This should give us a list of files that have not been
+      // assigned to an upload yet
+      const filesToUpload = waitingFileIDs.filter((fileId) => currentlyUploadingFiles.indexOf(fileId) < 0)
```

Comment:
Can we do this check a few lines up, here?

https://github.com/richardwillars/uppy/blob/0717ee8117a36510d582e960e223674cf98205a7/src/core/Core.js#L1083

`waitingFiles` was already supposed to be the list of files to upload. maybe we can just replace the `!file.uploadStarted` check on that line with `currentlyUploadingFiles.indexOf(fileId) === -1`?

Reply:
Ha, that was quite obvious.. how did I miss that?! Removed the filter loop etc and added the logic to the forEach. I kept the !file.progress.uploadStarted logic as without that it would start picking up files that have finished uploading